docs/v1/P4Runtime-Spec.mdk

```diff
@@ -511,6 +511,14 @@ In this situation, P4Info is selectively packaged into role-based subsets to
 allow some controllers access to just the functionality required. For example, a
 controller may only need read access to statistics counters and nothing more.
 
+## P4Runtime State Across Restarts { #sec-restarts }
+
+All targets support full restarts, where all forwarding state is reset and the
+P4Runtime server starts with a clean state.  Some targets may also support
+In-Service Software Upgrade (ISSU), where the software on the target can be
+restarted while traffic is being forwarded. In this case, the P4Runtime server
+may have the ability to access information from memory before the upgrade.
```

Comment:
sorry for the late comment, but "information from memory before the upgrade" sounds very vague to me. Shouldn't we clarify? Do you mean forwarding state, server state (such as the highest election id), or both?

Reply:
Waqar wanted to keep this relatively open, so that we don't restrict what exactly a server does.  The only requirement we have is the election ID needs to be preserved across ISSU, which we state clearly in the later section:

> This invariant must be maintained across in-service software upgrades, and the
> P4Runtime server must remember the highest `election_id` after such a restart.
> However, across a [full restart](#sec-restarts), the `election_id` must be
> reset. In fact, a full restart is the only way to reset the `election_id`.

Does that make sense?  I didn't want to duplicate the requirements in multiple places, and the later section seemed most appropriate for the actual requirement, but it didn't seem appropriate to introduce the notion of ISSU there.

Reply:
I don't have a strong objection. We can clarify later if needed.